src/sentry/monitors/logic/mark_failed.py

```diff
@@ -145,18 +146,16 @@ def mark_failed_threshold(failed_checkin: MonitorCheckIn, failure_issue_threshol
 
         starting_checkin = previous_checkins[0]
 
-        # for new incidents, generate a uuid as the fingerprint. This is
-        # not deterministic of any property of the incident and is simply
-        # used to associate the incident to it's event occurrences
-        fingerprint = uuid.uuid4().hex
-
-        MonitorIncident.objects.create(
+        incident, _ = MonitorIncident.objects.get_or_create(
             monitor=monitor_env.monitor,
```

Comment:
would this be able to create new monitor incidents even if one already exists (but is resolved) on the monitor environment?